python/ray/experimental/util/types.py

```diff
@@ -0,0 +1,19 @@
+from enum import Enum
+
+from ray.util.annotations import DeveloperAPI
+
+
+class _CollectiveOp(Enum):
+    pass
+
+
+@DeveloperAPI
+class ReduceOp(_CollectiveOp):
```

Comment:
No. Because the reduce op values are different in NCCL and Ray.

Reply:
`ray.util.collective.types`:
```python
class ReduceOp(Enum):
    SUM = 0
    PRODUCT = 1
    MIN = 2
    MAX = 3
```
actual NCCL ones:
```python
class ReduceOp(_CollectiveOp):
    SUM = 0
    PRODUCT = 1
    MAX = 2
    MIN = 3   # MIN, MAX are different
    AVG = 4
```

Reply:
I see, that looks like a bug. Instead of declaring a new enum, can you fix the old one and use it? That way we can also use approximately the same type signature for `.bind` vs `.__call__`.

Reply:
Updating an old file will need to update many other things enforced by the current CI. Should we proceed on updating `ray.util.collective.types`?

Reply:
Add a TODO and will fix later.